File: mns-core/src/main/java/cz/fb/manaus/core/provider/ExchangeProvider.java
package cz.fb.manaus.core.provider;

public class ExchangeProvider {

    private final String name;
    private final double minAmount;
    private final double minPrice;
    private final double chargeRate;
    private final boolean perMarketCharge;

    public ExchangeProvider(String name, double minAmount, double minPrice,
                            double chargeRate, boolean perMarketCharge) {
        this.name = name;
        this.minAmount = minAmount;
        this.minPrice = minPrice;
        this.chargeRate = chargeRate;
        this.perMarketCharge = perMarketCharge;
    }

    public double getMinAmount() {
        return minAmount;
    }

    public double getMinPrice() {
        return minPrice;
    }

    public double getChargeRate() {
        return chargeRate;
    }

    public boolean isPerMarketCharge() {
        return perMarketCharge;
    }

    public String getName() {
        return name;
    }

}

File: mns-core/src/main/java/cz/fb/manaus/core/provider/ExchangeProvider.kt
package cz.fb.manaus.core.provider

import java.util.*

data class ExchangeProvider(
        val name: String,
        val minAmount: Double,
        val minPrice: Double,
        val commission: Double,
        val tags: Set<ProviderTag>) {

    private val enumTags = EnumSet.copyOf(tags)

    fun validate() {
        validateTags(enumTags)
    }

    fun matches(selector: ProviderSelector) = enumTags.containsAll(selector.tags)
}


